internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -274,21 +274,34 @@ func (f *fleetGateway) gatherQueuedActions(ts time.Time) (queued, expired []flee
 	return queued, expired
 }
 
-func (f *fleetGateway) doExecute() (*fleetapi.CheckinResponse, error) {
+func (f *fleetGateway) executeCheckinWithRetries() (*fleetapi.CheckinResponse, error) {
 	f.backoff.Reset()
 
 	// Guard if the context is stopped by a out of bound call,
 	// this mean we are rebooting to change the log level or the system is shutting us down.
 	for f.bgContext.Err() == nil {
-		f.log.Debugf("Checkin started")
-		resp, err := f.execute(f.bgContext)
+		f.log.Debugf("Checking started")
+		resp, took, err := f.executeCheckin(f.bgContext)
 		if err != nil {
+			// Only update the local status on failure: https://github.com/elastic/elastic-agent/issues/1148
+			f.localReporter.Update(state.Degraded, fmt.Sprintf("checkin failed: %v", err), nil)
```

Comment:
I am wondering if we should only update the local reporter to the degraded state after multiple repeated failures, instead of just the first one?

In the fleet UI the agent is marked as degraded after multiple missed checkins, not just one.

Reply:
Sounds good, we can move it down to where it checks the fail counter

Reply:
Done